aea/helpers/dialogue/base.py

```diff
@@ -412,90 +395,145 @@ def last_message(self) -> Optional[Message]:
 
         return last_message
 
-    def get_message(self, message_id_to_find: int) -> Optional[Message]:
+    @property
+    def is_empty(self) -> bool:
+        """
+        Check whether the dialogue is empty.
+
+        :return: True if empty, False otherwise
+        """
+        return len(self._outgoing_messages) == 0 and len(self._incoming_messages) == 0
+
+    def counterparty_from_message(self, message: Message) -> Address:
+        if self.is_message_by_self(message):
+            counterparty = message.to
+        else:
+            counterparty = message.sender
+
+        return counterparty
+
+    def is_message_by_self(self, message: Message) -> bool:
+        """
+        Check whether the message is by this agent or not.
+
+        :param message: the message
+        :return: True if message is by this agent, False otherwise
+        """
+        return message.sender == self.agent_address
+
+    def is_message_by_other(self, message: Message) -> bool:
+        """
+        Check whether the message is by the counterparty agent in this dialogue or not.
+
+        :param message: the message
+        :return: True if message is by the counterparty agent in this dialogue, False otherwise
+        """
+        return not self.is_message_by_self(message)
+
+    def get_message(self, message_id: int) -> Message:
         """
         Get the message whose id is 'message_id'.
 
-        :param message_id_to_find: the id of the message
+        :param message_id: the id of the message
         :return: the message if it exists, None otherwise
         """
-        result = None  # type: Optional[Message]
+        assert not self.is_empty, "This dialogue is empty"
```

Comment:
Consistency in the error messages would be nice; i.e. always end with a dot for instance

Reply:
Done